File: iOS/todo-list/todo-list/Model/Action.swift
//
//  Activity.swift
//  todo-list
//
//  Created by Jason on 2022/04/07.
//

import Foundation

enum Action: String, Decodable {
    case Move = "MOVE"
    case Add = "ADD"
    case Delete = "DELETE"
    case Update = "UPDATE"
}

File: backend/src/main/java/com/hooria/todo/domain/Action.java
package com.hooria.todo.domain;

import java.util.Arrays;

public enum Action {

    UNKNOWN(0), ADD(1), REMOVE(2), UPDATE(3), MOVE(4);

    private final int code;

    Action(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public static Action of(int code) {
        return Arrays.stream(Action.values())
                .filter(d -> d.code == code)
                .findFirst()
                .orElse(UNKNOWN);
    }

    public static Action of(String device) {
        try {
            return Action.valueOf(device);
        } catch (IllegalArgumentException e) {
            return UNKNOWN;
        }
    }
}
